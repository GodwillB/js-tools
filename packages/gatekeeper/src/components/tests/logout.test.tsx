import session, { authenticateUser } from '@onaio/session-reducer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import { applyMiddleware, combineReducers, createStore, Store } from 'redux';
import { FlushThunks } from 'redux-testkit';
import thunk from 'redux-thunk';
import ConnectedLogout from '../logout';

describe('gatekeeper/ConnectedLogout', () => {
  let flushThunks;
  let store: Store;

  beforeEach(() => {
    flushThunks = FlushThunks.createMiddleware();
    store = createStore(combineReducers({ session }), applyMiddleware(flushThunks, thunk));
    jest.resetAllMocks();
  });

  it('renders the ConnectedLogout component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedLogout />
        </MemoryRouter>
      </Provider>
    );
    expect(toJson(wrapper.find('Connect(Logout)'))).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders the ConnectedLogout when logged in', () => {
    store.dispatch(
      authenticateUser(true, {
        email: 'bob@example.com',
        name: 'Bobbie',
        username: 'RobertBaratheon'
      })
    );
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedLogout />
        </MemoryRouter>
      </Provider>
    );
    expect(toJson(wrapper.find('Connect(Logout)'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
