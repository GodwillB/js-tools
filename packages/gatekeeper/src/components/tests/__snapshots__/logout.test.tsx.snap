// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatekeeper/ConnectedLogout renders the ConnectedLogout component when logged out 1`] = `
Object {
  "action": "REPLACE",
  "block": [Function],
  "canGo": [Function],
  "createHref": [Function],
  "entries": Any<Array>,
  "go": [Function],
  "goBack": [Function],
  "goForward": [Function],
  "index": 0,
  "length": 1,
  "listen": [Function],
  "location": ObjectContaining {
    "hash": "",
    "key": Any<String>,
    "pathname": "/login",
    "search": "",
    "state": undefined,
  },
  "push": [Function],
  "replace": [Function],
}
`;

exports[`gatekeeper/ConnectedLogout renders the ConnectedLogout component when logged out 2`] = `
<ProviderLinks
  providers={
    Object {
      "onadata": Object {
        "accessTokenUri": "https://stage-api.ona.io/o/token/",
        "authorizationUri": "https://stage-api.ona.io/o/authorize/",
        "clientId": "hunter2",
        "redirectUri": "https://example.com/oauth/callback/onadata/",
        "scopes": Array [
          "read",
          "write",
        ],
        "state": "abc",
        "userUri": "https://stage-api.ona.io/api/v1/user.json",
      },
    }
  }
>
  <div
    className="gatekeeper-login"
  >
    <p
      className="gatekeeper-p"
    >
      Please log in with one of the following providers
    </p>
    <p
      className="gatekeeper-p item"
      key="onadata"
    >
      <a
        className="gatekeeper-btn"
        href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=token&state=abc"
      >
        onadata
      </a>
    </p>
  </div>
</ProviderLinks>
`;

exports[`gatekeeper/ConnectedLogout renders the ConnectedLogout when logged in 1`] = `
Object {
  "action": "REPLACE",
  "block": [Function],
  "canGo": [Function],
  "createHref": [Function],
  "entries": Any<Array>,
  "go": [Function],
  "goBack": [Function],
  "goForward": [Function],
  "index": 0,
  "length": 1,
  "listen": [Function],
  "location": ObjectContaining {
    "hash": "",
    "key": Any<String>,
    "pathname": "/login",
    "search": "",
    "state": undefined,
  },
  "push": [Function],
  "replace": [Function],
}
`;

exports[`gatekeeper/ConnectedLogout renders the ConnectedLogout when logged in 2`] = `
<ProviderLinks
  providers={
    Object {
      "onadata": Object {
        "accessTokenUri": "https://stage-api.ona.io/o/token/",
        "authorizationUri": "https://stage-api.ona.io/o/authorize/",
        "clientId": "hunter2",
        "redirectUri": "https://example.com/oauth/callback/onadata/",
        "scopes": Array [
          "read",
          "write",
        ],
        "state": "abc",
        "userUri": "https://stage-api.ona.io/api/v1/user.json",
      },
    }
  }
>
  <div
    className="gatekeeper-login"
  >
    <p
      className="gatekeeper-p"
    >
      Please log in with one of the following providers
    </p>
    <p
      className="gatekeeper-p item"
      key="onadata"
    >
      <a
        className="gatekeeper-btn"
        href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=token&state=abc"
      >
        onadata
      </a>
    </p>
  </div>
</ProviderLinks>
`;

exports[`gatekeeper/ConnectedLogout works with its props as expected 1`] = `
Object {
  "action": "REPLACE",
  "block": [Function],
  "canGo": [Function],
  "createHref": [Function],
  "entries": Any<Array>,
  "go": [Function],
  "goBack": [Function],
  "goForward": [Function],
  "index": 0,
  "length": 1,
  "listen": [Function],
  "location": ObjectContaining {
    "hash": "",
    "key": Any<String>,
    "pathname": "/",
    "search": "",
    "state": undefined,
  },
  "push": [Function],
  "replace": [Function],
}
`;

exports[`gatekeeper/ConnectedLogout works with its props as expected 2`] = `
<div>
  Home
</div>
`;
